spark3/src/main/java/org/apache/iceberg/spark/procedures/ExpireSnapshotsProcedure.java

```diff
@@ -90,7 +96,14 @@ public StructType outputType() {
       if (retainLastNum != null) {
         action.retainLast(retainLastNum);
       }
-
+      if (numThreads != null && numThreads > 0) {
+        action.executeDeleteWith(Executors.newFixedThreadPool(numThreads, runnable -> {
```

Comment:
I'd probably move the initialization of a thread pool into a separate method that would accept a size.
Something similar to what @RussellSpitzer did in `BaseRewriteDataFilesSparkAction`.